src/lib/statistics/chunk_statistics/chunk_column_statistics.cpp

```diff
@@ -53,22 +53,18 @@ std::shared_ptr<ChunkColumnStatistics> ChunkColumnStatistics::build_statistics(D
         const auto& dictionary = *typed_column.dictionary();
         statistics = build_statistics_from_dictionary(dictionary);
     } else {
-      if constexpr(std::is_base_of_v<BaseEncodedColumn, ColumnType>) {
-        // if we have a generic encoded column we create the dictionary ourselves
-        auto iterable = create_iterable_from_column(typed_column);
-        std::unordered_set<DataTypeT> values;
-        iterable.for_each([&](const auto& value) {
-          // we are only interested in non-null values
-          if (!value.is_null()) {
-            values.insert(value.value());
-          }
-        });
-        pmr_vector<DataTypeT> dictionary{values.cbegin(), values.cend()};
-        std::sort(dictionary.begin(), dictionary.end());
-        statistics = build_statistics_from_dictionary(dictionary);
-      } else {
-       Fail("ChunkColumnStatistics should only be built for encoded columns.");
-      }
+      // if we have a generic encoded column we create the dictionary ourselves
```

Comment:
is this comment still correct? we are not only looking at encoded columns, are we?